tls/s2n_shutdown.c

```diff
@@ -33,14 +48,16 @@ int s2n_shutdown(struct s2n_connection *conn, s2n_blocked_status *blocked)
     POSIX_GUARD_RESULT(s2n_timer_elapsed(conn->config, &conn->write_timer, &elapsed));
     S2N_ERROR_IF(elapsed < conn->delay, S2N_ERR_SHUTDOWN_PAUSED);
 
-    /* Queue our close_notify alert.
-     * If we have already sent a close_notify, then this operation is a no-op.
-     */
-    POSIX_GUARD(s2n_queue_writer_close_alert_warning(conn));
-
-    /* Flush any pending records, then send the queued close_notify. */
+    /* Flush any outstanding data or alerts */
     POSIX_GUARD(s2n_flush(conn, blocked));
 
+    /* Send our close_notify alert if not sent yet */
```

Comment:
Theres some opportunity for some compliance comments here: https://www.rfc-editor.org/rfc/rfc8446#section-6.1
  "Each party MUST send a "close_notify" alert before closing its write
   side of the connection, unless it has already sent some error alert."